enzyme/Enzyme/CApi.cpp

```diff
@@ -340,6 +340,21 @@ void EnzymeRegisterAllocationHandler(char *Name, CustomShadowAlloc AHandle,
     };
 }
 
+
+/// This is the main entry point to register a custom derivative for language frontends.
+/// It should be prefered over trying to register custom-derivatives in the llvm-ir module.
+/// The main reason is that these rules can handle non-default activity cases, e.g.
+/// a function call where a pointer or a float scalar is marked as const.
+/// To get a better low-level understanding, the code in AdjointGenerator.h can be read.
+///
+/// This Function will only handle using ReverseMode AD (either split or combined).
+/// As a high-level example, assume we want to register a custom derivative for a vector resize function.
```

Comment:
I don't think this is a good example (far too complicated, and I'm not confident your algirthm for the derivative is correct). Perhaps explain with a simple pow(x, y) example where x or y could be differentiable or not and you want to handle both easily.

Reply:
Can you please actually explain why it is not correct?